resources/lib/cache.py

```diff
@@ -337,7 +337,14 @@ def verify_ttl(self, bucket, identifier, cache_entry):
             self._purge_entry(bucket, identifier)
             raise CacheMiss()
 
-    def _purge_entry(self, bucket, identifier):
+    def _purge_entry(self, bucket, identifier, on_disk=False):
+        # To ensure removing disk cache, it must be loaded first or it will trigger an exception
+        if on_disk:
+            try:
+                cache_entry = self._get_from_disk(bucket, identifier)
+                self.add(bucket, identifier, cache_entry['content'])
+            except:
+                pass
         # Remove from in-memory cache
```

Comment:
can you show me the error log?

Reply:
If there's no disk cache, I have a CacheMiss() error. I rewritten that function, you should like it better :)

Reply:
`# To ensure removing disk cache, it must be loaded first or it will trigger an exception
        cache_filename = self._entry_filename(bucket, identifier)
        cache_exixts = os.path.exists(cache_filename)

        if on_disk and cache_exixts:
            cache_entry = self._get_from_disk(bucket, identifier)
            self.add(bucket, identifier, cache_entry['content'])

        # Remove from in-memory cache
        del self._get_bucket(bucket)[identifier]
        # Remove from disk cache if it exists

        if cache_exixts:
            os.remove(cache_filename)`